### PR TITLE
Use new gene pane endpoint; Progress bar update

### PR DESCRIPTION
**Pull request by zhx828**
Use latest gene panel endpoint to improve the performance;
Cover gene panel endpoints into progress bar;
Fake the progress base on the time the component has been loaded;

### CODE REVIEW

File: app/scripts/model/dataProxy.js [hunk 1]
```diff
@@ -1367,102 +1373,95 @@ window.DataManagerForIviz = (function($, _) {
...
+                  _.each(panel.genes, function(gene) {
+                    if (!geneSampleMap[gene.hugoGeneSymbol]) {
+                      geneSampleMap[gene.hugoGeneSymbol] = {
+                        sampleUids: []
+                      }
+                    }
+                    geneSampleMap[gene.hugoGeneSymbol].sampleUids = geneSampleMap[gene.hugoGeneSymbol].sampleUids.concat(panelSamplesMap[panel.genePanelId]);
+                  })
+                });
+                if (panelSamplesMap.hasOwnProperty(wholeExomeSequencedPanelId)) {
```
Comment (kalletlak) [inline app/scripts/model/dataProxy.js:1425]: may be you could move this condition after line 1422

File: app/scripts/model/dataProxy.js [hunk 2]
```diff
@@ -1367,102 +1373,95 @@ window.DataManagerForIviz = (function($, _) {
...
+              contentType: "application/json; charset=utf-8",
+              success: function(panels) {
+                _.each(panels, function(panel) {
+                  _.each(panel.genes, function(gene) {
+                    if (!geneSampleMap[gene.hugoGeneSymbol]) {
+                      geneSampleMap[gene.hugoGeneSymbol] = {
+                        sampleUids: []
+                      }
+                    }
+                    geneSampleMap[gene.hugoGeneSymbol].sampleUids = geneSampleMap[gene.hugoGeneSymbol].sampleUids.concat(panelSamplesMap[panel.genePanelId]);
```
Comment (kalletlak) [inline app/scripts/model/dataProxy.js:1422]: is there any possibility for duplicate ids

File: app/scripts/model/dataProxy.js [hunk 3]
```diff
@@ -1367,102 +1373,95 @@ window.DataManagerForIviz = (function($, _) {
...
+                  })
+                });
+                if (panelSamplesMap.hasOwnProperty(wholeExomeSequencedPanelId)) {
+                  _.each(geneSampleMap, function(item) {
+                    item.sampleUids = item.sampleUids.concat(panelSamplesMap[wholeExomeSequencedPanelId]);
```
Comment (kalletlak) [inline app/scripts/model/dataProxy.js:1427]: even here, any duplicate ids

File: app/scripts/model/dataProxy.js (hunk 1 above)
Comment (zhx828) [reply app/scripts/model/dataProxy.js:1425]: Nope, you don't want to do a O(n2) if O(1) works here.

File: app/scripts/model/dataProxy.js (hunk 2 above)
Comment (zhx828) [reply app/scripts/model/dataProxy.js:1422]: I don't think so. All they all uids. And panels cover different samples.

File: app/scripts/model/dataProxy.js (hunk 3 above)
Comment (zhx828) [reply app/scripts/model/dataProxy.js:1427]: Nope. Samples in wholeExomeSequeced are without any panel info.

**Review by kalletlak — APPROVED** [review]

**Pull request merged**